scripts/ci/utils/update-dcos-repo

```diff
@@ -0,0 +1,82 @@
+#!/bin/bash
+
+set -e
+[ -n "${DEBUG}" ] && set -x
+
+# This script is intended to be called by the ci scripts, please dont 
+# execute it directly.
+
+## Configuration
+#####################################################################
+
+# path of this file
+SCRIPT_PATH="$(cd $(dirname "$0")/$(dirname "$(readlink "$0")") && pwd)"
+
+# project root for this file
+PROJECT_ROOT="$( cd "$( echo ${SCRIPT_PATH} | sed s+/scripts/ci/utils++)" && pwd )"
+
+# where to store dcos repo in
+DCOS_FOLDER=${DCOS_FOLDER:-"dcos-repo"}
+
+# DCOS Origin and Upstream
+DCOS_ORIGIN=${DCOS_ORIGIN:-"https://${GIT_USER}:${GIT_PASSWORD}@github.com/mesosphere/dcos"}
+DCOS_UPSTREAM=${DCOS_UPSTREAM:-"https://github.com/dcos/dcos"}
+
+# these are passed in from calling script
+PKG_VERSION=${PKG_VERSION:-""}
+PKG_TARGET=${PKG_TARGET:-""}
+DCOS_BRANCH=${DCOS_BRANCH:-""}
+
+DCOS_BASE_BRANCH=${DCOS_BASE_BRANCH:-"$(echo ${PKG_TARGET})"}
+
+## Check Params
+#####################################################################
+
+if [ -z "${DCOS_BRANCH}" ] || [ -z "${PKG_TARGET}" ] || [ -z "${PKG_VERSION}" ]; then
+  echo "Cant update DC/OS repo without given \$DCOS_BRANCH, \$PKG_TARGET and \$PKG_VERSION!"
+  exit 1
+fi
+
+## Update DC/OS Repo
+#####################################################################
+
+[ -d "${PROJECT_ROOT}/${DCOS_FOLDER}" ] && rm -rf ${PROJECT_ROOT}/${DCOS_FOLDER}
+
+# clone & checkout correct branch
+git clone "${DCOS_ORIGIN}" ${PROJECT_ROOT}/${DCOS_FOLDER}
+
+git -C "${PROJECT_ROOT}/${DCOS_FOLDER}" \
+  checkout -B ${DCOS_BRANCH}
+
+# rebase onto latest master
+git -C "${PROJECT_ROOT}/${DCOS_FOLDER}" \
+  remote add upstream "${DCOS_UPSTREAM}"
+
+# rebase onto latest master
+git -C "${PROJECT_ROOT}/${DCOS_FOLDER}" \
+  fetch upstream
+
+git -C "${PROJECT_ROOT}/${DCOS_FOLDER}" \
+  reset --hard upstream/${DCOS_BASE_BRANCH}
+
+# copy previously created files
+if [ -f "${PROJECT_ROOT}/buildinfo.json" ]; then
```

Comment:
☝️ We should fail if `buildinfo.json` is not present.

Reply:
✅